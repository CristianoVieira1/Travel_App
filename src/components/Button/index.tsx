import ImageBtn from "@assets/images/btn.png";
import { Load } from "@components/Animations/Load";
import { AntDesign as Icon } from "@expo/vector-icons";
import React, { memo } from "react";
import { TouchableOpacityProps } from "react-native";
import { useTheme } from "styled-components/native";
import * as S from "./styles";

interface IButtonProps extends TouchableOpacityProps {
  title: string | undefined;
  loading?: boolean;
  type: S.TypeButton;
  size?: S.SizeButton;
  fontSize?: S.SizeButton;
  icon?: string;
}

function Button({
  title,
  loading = false,
  type,
  size,
  icon,
  ...rest
}: IButtonProps) {
  const theme = useTheme();
  return (
    <S.Container
      {...rest}
      activeOpacity={0.7}
      type={type}
      size={size}
      icon={icon}
      style={type !== "link" && { ...theme.shadow }}
    >
      {loading ? (
        <Load />
      ) : (
        <>
          <S.Content>
            <S.Image source={ImageBtn} type={type} size={size} />
            <S.TitleButton type={type}>{title}</S.TitleButton>

            <S.Icon>
              <Icon name={icon} size={24} color="white" />
            </S.Icon>
          </S.Content>
        </>
      )}
    </S.Container>
  );
}
export default memo(Button);
