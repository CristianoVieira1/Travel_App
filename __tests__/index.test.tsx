import { render } from "@testing-library/react-native";
import React from "react";
import Button from "../src/components/Button";
import { withTheme } from "../src/theme/withTheme";

describe("Button", () => {
  it("The component rendered", () => {
    const snapshot = render(
      withTheme(
        <Button
          title="Teste"
          fontSize="medium"
          loading={false}
          size="medium"
          type="accept"
          icon={"home"}
          onPress={() => {}}
        />
      )
    ).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
