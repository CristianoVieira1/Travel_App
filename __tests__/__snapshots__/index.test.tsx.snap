// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button snapshots 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 8,
      "elevation": 5,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "shadowColor": "#00000066",
      "shadowOffset": Object {
        "height": 3,
        "width": 5,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
      "width": "50%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      size="medium"
      source={1}
      style={
        Array [
          Object {
            "borderRadius": 8,
            "display": "flex",
            "width": "95%",
          },
        ]
      }
      type="accept"
    />
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "PublicSans_500Medium",
            "fontSize": 16,
            "position": "absolute",
          },
        ]
      }
      type="accept"
    >
      Teste
    </Text>
    <View
      style={
        Array [
          Object {
            "marginLeft": 10,
            "marginTop": 5,
          },
        ]
      }
    >
      <Text />
    </View>
  </View>
</View>
`;
